### DISCUSSION TITLE
CORS Rules / ZipArchive::Close cannot open file

### DISCUSSION THREAD

**Discussion by jasperf** [spatie/laravel-backup#1345] [Q&A]
Been having issues with removing old backups

```
Important: An error occurred while backing up App
Exception message: ZipArchive::close(): Can't open file: No such file or directory
Exception trace: #0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError() #1 
```

This occurs every now and then. Started to wonder about general CORS rules to set on Digital Ocean Spaces . Do we need to allow `GET, PUT, DELETE, POST, HEAD` for the domain sending backups to Digital Ocean Spaces to have Laravel Backup work properly?

Two, does the `ZipArchive::Close cannot open file: No Such File or Directory` error have anything to do with this?

Here our `backup.php exclusion list`:

```
/*
* These directories and files will be excluded from the backup.
*
* Directories used by the backup process will automatically be excluded.
*/
'exclude' => [
base_path('vendor'),
base_path('node_modules'),
base_path('.git'),
base_path('public/published'),
base_path('public/uploads'),
],
```

Should I perhaps exclude other data like session data like mentioned at https://github.com/spatie/laravel-backup/issues/138 ?

**Comment by jasperf** [comment]
Also see https://github.com/spatie/laravel-backup/discussions/1325 . It seems I should exclude the framework directory entirely including cache, sessions and views. So seems it was not an issue of general CORS rules so removed these for now and updated the Laravel backup exclusion list with:

```
base_path('storage/framework'),
```
.

**Reply by jonesheckler** [reply]
Thank you very much

This helped me a lot wih my error...


Exception message: ZipArchive::close(): Can't open file: No such file or directory
Exception trace: #0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError() #1
